README.md

```diff
@@ -57,10 +112,11 @@ function Err<Err>(err: Err): ERR<Err>;
 // Convert exceptions into errors & function result into Ok
 function toResult<Val, ERR>(fn: () => Val): Result<Val, ERR>;
 
-// Convert promise reject into errors & promise resolve into Ok
-async function toResultAsync<Val, ERR>(fn: () => Promise<Val>): Promise<Result<Val, ERR>>;
+// Convert function returning T and error mapper returning E to new function returning Result<T, E>
+function resultify<T, E, F extends Fn<T>>(fn: F, mapErr: (err: unknown) => E): ResFn<T, E, F>;
 
-type AsyncResult<Val, ERR> = Promise<Result<Val, ERR>>;
```

Comment:
Typing `Promise<Result<Val, Err>>` was not cool, too many generic layers. We can't return `AsyncResult` instance directly from async function. So the type is still needed. Just change it to `AsyncResultT`.

```
type AsyncResultT<Val, Err = Error> = Promise<AsyncResult<Val, Err>>;
```

Reply:
I would like to not override meaning of basically same name to two quite different things. How does "ResultPromise" for `Promise<Result<Val, Err>>` and "AsyncResultPromise" `Promise<AsyncResult<Val, Err>>` sound to you?

Reply:
I think `ResultPromise` and `AsyncResultPromise` are great

Reply:
Added